### PR TITLE
Revert 15398

### PR DESCRIPTION
**Pull request by ycliuhw**
Revert https://github.com/juju/juju/pull/15398

Conflicts:
```
- apiserver/admin.go
- apiserver/apiserver.go
- apiserver/common/crossmodel/auth.go
- apiserver/export_test.go
- apiserver/facade/interface.go
- apiserver/facades/client/applicationoffers/applicationoffers.go
- apiserver/facades/client/applicationoffers/applicationoffers_test.go
- apiserver/facades/controller/crossmodelrelations/crossmodelrelations.go
- apiserver/localofferauth.go
- apiserver/root.go
- apiserver/testing/fakeauthorizer.go

```

### CODE REVIEW

File: apiserver/facades/client/applicationoffers/applicationoffers.go [hunk 1]
```diff
@@ -462,34 +462,6 @@ func (api *OffersAPI) getConsumeDetails(user names.UserTag, urls params.OfferURL
...
-		isAdmin := err == nil
-		if !isAdmin {
-			appOffer := names.NewApplicationOfferTag(offer.OfferUUID)
-			err := api.Authorizer.EntityHasPermission(user, permission.ConsumeAccess, appOffer)
-			if err != nil {
-				results[i].Error = apiservererrors.ServerError(err)
-				continue
-			}
-		}
-
```
Comment (wallyworld) [inline apiserver/facades/client/applicationoffers/applicationoffers.go:492]: keep this

File: apiserver/testing/fakeauthorizer.go [hunk 2]
```diff
@@ -152,13 +150,5 @@ func (fa FakeAuthorizer) EntityHasPermission(entity names.Tag, operation permiss
 	if fa.AdminTag != emptyTag && entity == fa.AdminTag {
 		return nil
 	}
-	if operation == permission.ConsumeAccess && fa.HasConsumeTag != emptyTag && entity == fa.HasConsumeTag {
```
Comment (wallyworld) [inline apiserver/testing/fakeauthorizer.go:155]: We might still need this, not sure. See if the tests fail.

**Review by wallyworld — APPROVED** [review]
A few changes to keep.

**Comment by manadart** [comment]
What motivated the need to revert?

**Pull request merged**